### PR TITLE
Change gen JSON output from stream to array

### PR DESCRIPTION
**Pull request by vinamogit**
### What

`soroban gen --wasm ... --output json` gives an invalid json output

### Why

Fix the json output of the `gen` command

### Known limitations

NA

### CODE REVIEW

File: soroban-spec/src/gen/json.rs
Comment (leighmcculloch) [inline soroban-spec/src/gen/json.rs:59]: The current output is a valid JSON stream where each entity is a JSON object. Many tools will support it, although for some it might be inconvenient. For example, you can pipe the stream to tools like `jq` to operate on it.

What situations are you seeing this as invalid?

Making it a JSON array seems reasonable way of increasing interop.

Comment (vinamogit) [reply soroban-spec/src/gen/json.rs:59]: I was trying to import the output in javascript and the JSON parse complains.

File: soroban-spec/src/gen/json.rs
Comment (leighmcculloch) [inline soroban-spec/src/gen/json.rs:54]: Could you change building the JSON manually here, to doing a collect into a Vec<_>, then calling serde_json::to_string_pretty on that vec? It will have us continue to rely on the serde_json crate for serializing and formatting all the JSON consistently. This change as it is results in unformatted.

Alternatively, if you check the box on the PR that lets maintainers push commits I'll push the change and merge it.

**Review by leighmcculloch — APPROVED** [review]
Thanks @vinamogit !!!

**Comment by leighmcculloch** [comment]
@vinamogit This can't be merged until you merge the `main` branch into your branch.

**Comment by leighmcculloch** [comment]
@vinamogit Your branch is out of date again. Could you check the box that lets me be able to push to the branch, so that I can update it and merge it in one step?

**Comment by vinamogit** [comment]
> 

Sorry, can't find the option. I've added you as contributor the the repo, hope that helps.

**Comment by leighmcculloch** [comment]
It's merged! Thanks @vinamogit. You can remove me now.

**Pull request merged**
